rustls/src/crypto/signer.rs

```diff
@@ -186,7 +194,7 @@ pub struct SelectedCredential {
 
 /// A peer's identity, depending on the negotiated certificate type.
 #[non_exhaustive]
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
```

Comment:
Have been thinking: should we push `Identity` (et al) down into pki-types? I know there are some downstream crates that expose an identity type for client authentication.